.clang-tidy

```diff
@@ -100,8 +84,128 @@ Checks: >-
   -readability-magic-numbers,
   -readability-named-parameter,
   -readability-suspicious-call-argument,
-  -COMMENT---aliases-of-above---,
-  -cert-str34-c
+  -COMMENT---disable-all-aliases---,
+  -cert-con36-c,
+  -cert-con54-cpp,
+  -cert-dcl03-c,
+  -cert-dcl16-c,
+  -cert-dcl37-c,
+  -cert-dcl51-cpp,
+  -cert-dcl54-cpp,
+  -cert-dcl59-cpp,
+  -cert-err09-cpp,
+  -cert-err61-cpp,
+  -cert-exp42-c,
+  -cert-fio38-c,
+  -cert-flp37-c,
+  -cert-msc30-c,
+  -cert-msc32-c,
+  -cert-oop11-cpp,
+  -cert-oop54-cpp,
+  -cert-pos44-c,
+  -cert-pos47-c,
+  -cert-sig30-c,
+  -cert-str34-c,
+  -clang-analyzer-core.CallAndMessage,
```

Comment:
I am pretty much sure we are not running the Clang Static Analyzer as it is a separate tool and we never got any errors for the checks of clang-analyzer* we had disabled before from the static analyzer. 
So please revert all changes for the clang-analyzer* checks. 
We just need to ensure not to enable the static Analyzer checks

Reply:
Actually the log says that they were still enabled, so it looks like they can't be disabled? So I'll take them out, to avoid misleading readers.